File: examples/src/Presentations/Properties/CheckPresentationProtection.cs
using System
using System.Collections.Generic
using System.IO
using System.Linq
using System.text
using System.Threading.Tasks
import aspose.slides as slides
using Aspose.slides.Examples.CSharp
using Aspose.slides.Export
using Aspose.slides.Util

namespace CSharp.Presentations.properties
{
    class CheckPresentationProtection
    {
        public static void Run()
        {
            #Path for source presentation
            pptxFile = Path.Combine(RunExamples.GetDataDir_PresentationProperties(), "modify_pass2.pptx")
            pptFile = Path.Combine(RunExamples.GetDataDir_PresentationProperties(), "open_pass1.ppt")

            # Check the Write Protection Password via IPresentationInfo Interface
            IPresentationInfo presentationInfo = PresentationFactory.Instance.GetPresentationInfo(pptxFile)
            bool isWriteProtectedByPassword = presentationInfo.IsWriteProtected == 1 && presentationInfo.CheckWriteProtection("pass2")
            print("Is presentation write protected by password = " + isWriteProtectedByPassword)

            # Check the Write Protection Password via IProtectionManager Interface
            using (presentation = new Presentation(pptxFile))
            {
                bool isWriteProtected = presentation.ProtectionManager.CheckWriteProtection("pass2")
                print("Is presentation write protected = " + isWriteProtected)
            }

            # Check Presentation Open Protection via IPresentationInfo Interface
            presentationInfo = PresentationFactory.Instance.GetPresentationInfo(pptFile)
            if (presentationInfo.IsPasswordProtected)
            {
                print("The presentation '" + pptxFile + "' is protected by password to open.")
            }
        }
    }
}

File: examples/src/Presentations/properties/check_presentation_protection.py
import aspose.slides as slides


def props_check_presentation_protection(global_opts):
    # Check the Write Protection Password via IPresentationInfo Interface
    presentation_info = slides.PresentationFactory.instance.get_presentation_info(
        global_opts.data_dir + "props_check_presentation_protection.pptx")
    is_write_protected_by_password = presentation_info.is_write_protected == slides.NullableBool.TRUE and \
        presentation_info.check_write_protection("pass2")
    print("Is presentation write protected by password = " + str(is_write_protected_by_password))

    # Check the Write Protection Password via IProtectionManager Interface
    with slides.Presentation(global_opts.data_dir + "props_check_presentation_protection.pptx") as presentation:
        is_write_protected = presentation.protection_manager.check_write_protection("pass2")
        print("Is presentation write protected = " + str(is_write_protected))

    # Check Presentation Open Protection via IPresentationInfo Interface
    presentation_info = slides.PresentationFactory.instance.get_presentation_info(
        global_opts.data_dir + "props_ppt_with_password.ppt")
    if presentation_info.is_password_protected:
        print("The presentation 'props_ppt_with_password.ppt' is protected by password to open.")
